File: Assets/Scripts/MainGameScript.js
public var car : GameObject;
public var CarSpawnDelay : float;

var readynow : boolean = true;
// http://www.unity3dstudent.com/2010/09/beginner-b22-pausing-scripts-with-waitforseconds/

function Update () {
	if(readynow){
  		CreateCars();
  	} 
}

function CreateCars() {
	
	readynow=false;
	
	// create a new car
	var track = Random.Range(0,-4);
	Debug.Log(track);
	var car = Instantiate(car,Vector3(8,2,(track*4)), Quaternion.identity);
   
   	// create timer delay
	var waitTime : float = Random.Range(0.1,CarSpawnDelay);
	yield WaitForSeconds(waitTime);
	
	readynow=true;
}

File: Assets/Scripts/MainGameScript.cs
using UnityEngine;
using System.Collections;

public class MainGameScript : MonoBehaviour {
	public GameObject car;
	public float CarSpawnDelay;

	bool  readynow = true;
	// http://www.unity3dstudent.com/2010/09/beginner-b22-pausing-scripts-with-waitforseconds/

	void  Update (){
		if(readynow){
            StartCoroutine(CreateCars());
		} 
	}

		IEnumerator  CreateCars (){

		readynow=false;

		// create a new car
		float track= Random.Range(0,-4);
		Debug.Log(track);
        GameObject newcar = (GameObject)Instantiate(car, new Vector3(8,2,(track*4)), Quaternion.identity);

		// create timer delay
		float waitTime = Random.Range(0.1f,CarSpawnDelay);
		yield return new WaitForSeconds(waitTime);

		readynow=true;
	}
}